R/derive_var_trtemfl.R

```diff
@@ -0,0 +1,237 @@
+#'
+#' @param trt_start_date Treatment start date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset
+#'
+#' @param trt_end_date Treatment end date
+#'
+#'   *Permitted Values:* A symbol referring to a date or datetime variable of
+#'   the input dataset or `NULL`
+#'
+#' @param end_window
+#'
+#'    If the argument is specified, events starting more than the specified
+#'    number of days after end of treatment, are not flagged.
+#'
+#'    *Permitted Values:* A non-negative integer or `NULL`
+#'
+#' @param ignore_time_for_trt_end
+#'
+#'   If the argument is set to `TRUE`, the time part is ignored for checking if
+#'   the event occurred more than `end_window` days after end of treatment.
+#'
+#'   *Permitted Values:* `TRUE`, `FALSE`
+#'
+#' @param initial_intensity Initial severity/intensity or toxity
+#'
+#'   If the argument is specified, events which start before treatment start and
+#'   end after treatment start (or are ongoing) and worsened (i.e., the
+#'   intensity is greater than the initial intensity), are flagged.
+#'
+#'   The values of the specified variable must be comparable with the usual
+#'   comparison operators. I.e., if the intensity is greater than the initial
+#'   intensity `initial_intensity < intensity` must evaluate to `TRUE`.
+#'
+#'   *Permitted Values:* A symbol referring to a variable of the input dataset
+#'   or `NULL`
+#'
+#' @param intensity Severity/intensity or toxity
+#'
+#'   If the argument is specified, events which start before treatment start and
+#'   end after treatment start (or are ongoing) and worsened (i.e., the
+#'   intensity is greater than the initial intensity), are flagged.
+#'
+#'   The values of the specified variable must be comparable with the usual
+#'   comparison operators. I.e., if the intensity is greater than the initial
+#'   intensity `initial_intensity < intensity` must evaluate to `TRUE`.
+#'
+#'   *Permitted Values:* A symbol referring to a variable of the input dataset
+#'   or `NULL`
+#'
+#' @author Stefan Bundfuss
+#'
+#' @details For the derivation of the new variable the following cases are
+#'   considered in this order. The first case which applies, defines the value
+#'   of the variable.
+#'
+#' - *not treated*: If `trt_start_date` is `NA`, it is set to `NA_character_`.
+#' - *event before treatment*: If `end_date` is before `trt_start_date` (and
+#' `end_date` is not `NA`), it is set to `NA_character_`.
+#' - *no event date*: If `start_date` is `NA`, it is set to `"Y"`.
+#' - *event started during treatment*:
+#'     - if `end_window` is not specified:
+#'       if `start_date` is on or after `trt_start_date`, it is set to `"Y"`,
+#'     - if `end_window` is specified:
+#'       if `start_date` is on or after `trt_start_date` and `start_date` is on
+#'       or before `trt_end_date` + `end_window` days, it is set to `"Y"`,
+#'  - *event started before treatment and (possibly) worsened on treatment*: If
```

Comment:
i think there should be a space and a dash before the "If..." so that this gets a new line similar to the above section

Reply:
Done